flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java

```diff
@@ -812,18 +854,16 @@ public JobExecutionResult execute(String jobName) throws Exception {
 
 		final Plan plan = createProgramPlan(jobName);
 		final ExecutorFactory executorFactory =
-				executorServiceLoader.getExecutorFactory(configuration);
-
-		final Executor executor = executorFactory.getExecutor(configuration);
+			executorServiceLoader.getExecutorFactory(configuration);
 
-		try (final JobClient jobClient = executor.execute(plan, configuration).get()) {
+		checkNotNull(
+			executorFactory,
+			"Cannot find compatible factory for current environment(%s)",
+			getClass().getSimpleName());
 
-			lastJobExecutionResult = configuration.getBoolean(DeploymentOptions.ATTACHED)
-					? jobClient.getJobExecutionResult(userClassloader).get()
-					: new DetachedJobExecutionResult(jobClient.getJobID());
+		final Executor executor = executorFactory.getExecutor(configuration);
 
-			return lastJobExecutionResult;
-		}
+		return executor.execute(plan, configuration);
```

Comment:
This is just up to you to integrate if you like it, but I would inline the methods:

```
return executorFactory
				.getExecutor(configuration)
				.execute(plan, configuration);
```

Reply:
OK I'll integrate it.